[2단계 - 자동차 경주 리팩토링] 루쿠(백경환) 미션 제출합니다

aiaiaiai1 opened the pull request:
라빈 안녕하세요
이번 1단계 리뷰를 통해서 단위 테스테를 작성할때 저만의 기준과 방향을 잡는데 도움이 많이 되었습니다. 감사합니다!

이번 리펙토링 단계에서는 설계와 관련해서 도메인과 뷰의 분리를 개선하고자 하는데 이를 위해서 조금의 힌트가 필요한거같아 설계에있어 제가 놓치거나 아쉬웠던 부분에 대해서 조심스럽게 질문 드리고자합니다 감사합니다

Commit:
refactor: 보다 직관적으로 모델과 뷰를 분리하는 거에 대한 수정

Commit:
feat: 배열을 리스트로 변경

Commit:
test: 배열을 리스트로 변경

Commit:
refactor: 하드코딩 삭제

giantim reviewed (changes requested):
루쿠 안녕하세요~
2단계 미션 빠르게 작업해주셨네요! 질문주신 내용과 관련해서 한번에 코멘트 남기도록 하겠습니다 😉
- 지금 설계에서도 충분히 요구사항을 잘 만족하는 프로그램을 작성해주신거 같아요. 그럼 이제 더 객체의 역할과 책임을 엄밀하게 분리해볼까요? Car 클래스가 이동 여부를 판단하는 책임을 갖지 않게 만들어봐도 좋을것 같아요.
- 이번 설계에서 도메인과 뷰를 나누는 것에 대해서 정말 많은 고민을 하셨었고 관련해서 저도 리뷰를 남겼었는데요. 저는 아직 view와 handler의 명확한 구분을 모르겠네요 😄 입력값의 유효성 검증은 InputHandler가 담당하는데 출력시 사용자에게 보여줘야 하는 형식은 OutputView에서 담당하고 있네요? view 패키지에 대해 엄밀하게 정의해보면 좋을것 같아요
- 리팩토링 과정에서 프로그램의 예외 처리에 대해 바꿔보면 어떨까요? 현재 입력값이 잘못됐을 시 프로그램이 종료되는데 이 부분을 개선해 보면서 컨트롤러 단에서 예외를 어떻게 처리할지 고민해봐도 좋을것 같아요.
- 커밋 메시지를 남길때 좀 더 구체적으로 적어보면 좋을것 같아요. 다른 개발자가 봤을 때 어느 부분을 개선한건지 이해하기 어렵습니다!

Commit:
refactor: InputHandler 클래스 삭제

- InputHandler 클래스를 InputView 클래스로 통합
- RancingGameController에서 InputHandler 대신 InputView를 통해서 입력값 받기

Commit:
feat: 입력값이 빈 문자열인 경우 예외 발생 기능 추가

+ public 메서드 위치 변경

Commit:
feat: 자동차 이름에 대한 입력값이 잘못된경우 재입력 받는 기능 추가

Commit:
style: 자바 컨벤션 수정

aiaiaiai1 commented:
라빈 안녕하세요
말씀해주신것처럼 view와 handler의 명확한 구분이 없고 일관성이 많이 떨어지는 것 같습니다... 객체의 역할과 책임을 엄밀하게 분리하는거에 대해서도 좀 더 고민해봐야겠지만 객체 입장과 관점으로 좀더 생각해볼수있는 계기가 된것 같습니다! 좋은고민거릴 남겨주신 답변과 피드백 감사합니다!!

giantim reviewed (approved):
루쿠 안녕하세요!
이번 리팩토링 단계도 빠르게 잘 마무리 해주셨네요! 고생 많으셨습니다. 다음 미션도 화이팅입니다~
